pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -717,16 +744,29 @@ func (a *FlavorAssigner) findFlavorForPodSets(
 		attemptedFlavorIdx = idx
 		fName := resourceGroup.Flavors[idx]
 
-		if fit, err := a.checkFlavorForPodSets(log, fName, psIDs, podSets, selectors, status); !fit {
+		ft := FlavorAssignmentAttempt{Flavor: fName}
+		flavorStatus := NewStatus()
+
+		if fit, err := a.checkFlavorForPodSets(log, fName, psIDs, podSets, selectors, flavorStatus); !fit {
+			ft.Mode = NoFit
+			if flavorStatus != nil {
+				ft.Reasons = append(ft.Reasons, flavorStatus.reasons...)
+				status.reasons = append(status.reasons, flavorStatus.reasons...)
+			}
+			considered = append(considered, ft)
```

Comment:
TBH this is a bit complicated, maybe we could introduce a helper type like "FlavorAssignmentAttempts". Then, in the "flavor_assignment_attempts.go" for example, and it could encapsulate the serialization logic which now is in the main scheduler.go, plus all the other helper logic